subjects/05-props-v-state/exercise.js

```diff
@@ -75,13 +71,62 @@ const Tabs = React.createClass({
 
 })
 
+const StatefulTabs = React.createClass({
+  getInitialState() {
+    return {
+      tabIndex: 0
+    }
+  },
+
+  tabChanged(tabIndex) {
+    this.setState({
+      tabIndex: tabIndex
+    })
+  },
+
+  render() {
+    return <Tabs {...this.props} onTabChange={this.tabChanged} activeIndex={this.state.tabIndex} />
+  }
+});
+
 const App = React.createClass({
+  getInitialState() {
+    return {
+      tabIndex: 0
+    }
+  },
+
+  tabChanged(tabIndex) {
+    this.setState({
+      tabIndex: tabIndex
+    })
+  },
+
+  decrementTab() {
+    this.setState({
+      tabIndex: this.state.tabIndex - 1
+    })
+  },
+
+  incrementTab() {
+    this.setState({
+      tabIndex: this.state.tabIndex + 1
+    })
+  },
 
   render() {
     return (
       <div>
         <h1>Props v. State</h1>
-        <Tabs ref="tabs" data={this.props.tabs} />
+
+        {this.state.tabIndex > 0 ? <button onClick={this.decrementTab}>rewind</button> : ''}<br />
+        {this.state.tabIndex < this.props.tabs.length - 1 ? <button onClick={this.incrementTab}>fast forward</button> : ''}
+
+        <h2>The stateless one (controllable above also)</h2>
```

Comment:
`Tabs` passes state back up, but the FF/Rewind buttons above can push it back down, think it's closer to some of our use cases